### PR TITLE
Add the ineligible post processor to the log recovery proposal flow

### PR DESCRIPTION
**Pull request by ferglor**
In this PR, we're adding the ineligible post processor to the log recovery proposal flow

We're also fixing some flaky tests (race conditions), and skipping two recoverable service tests that can occasionally become deadlocked

### CODE REVIEW

File: pkg/v3/postprocessors/ineligible_test.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+					Eligible:               true,
+				},
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			expectsErr: true,
+			wantErr: errors.New(`upkeep state boom
+upkeep state boom`),
```
Comment (infiloop2) [inline pkg/v3/postprocessors/ineligible_test.go:86]: 💥

File: pkg/v3/postprocessors/ineligible_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+					return nil
+				},
+			},
+			results: []types.CheckResult{
+				{
+					PipelineExecutionState: 0,
+					Eligible:               false,
+				},
+			},
+			wantLog: "post-processing 1 results, 0 ineligible",
```
Comment (infiloop2) [inline pkg/v3/postprocessors/ineligible_test.go:38]: should the log here indicate 1 ineligible?

Comment (infiloop2) [reply pkg/v3/postprocessors/ineligible_test.go:38]: ah i think there's a bug in ineligible processor when counting ineligible count

**Review by infiloop2 — CHANGES_REQUESTED** [review]
Requesting changes to investigate ineligible processor log

File: pkg/v3/postprocessors/ineligible_test.go (hunk 1 above)
Comment (ferglor) [reply pkg/v3/postprocessors/ineligible_test.go:86]: This is a multi error check, so the new line is needed

Comment (infiloop2) [reply pkg/v3/postprocessors/ineligible_test.go:86]: i was just pointing out the boom :p

File: pkg/v3/postprocessors/ineligible_test.go (hunk 2 above)
Comment (ferglor) [reply pkg/v3/postprocessors/ineligible_test.go:38]: Should the ineligible loop then become:

```if res.PipelineExecutionState == 0 && !res.Eligible {
	err := p.stateUpdater.SetUpkeepState(ctx, res, ocr2keepers.Ineligible)
	if err != nil {
		merr = errors.Join(merr, err)
		continue
	}
	ineligible++
}```

Comment (infiloop2) [reply pkg/v3/postprocessors/ineligible_test.go:38]: yup i think so!

**Pull request merged**